### ISSUE TITLE
0.3.0 Backpack Won't Open Around Other Players in Multiplayer

### ISSUE THREAD

**Issue by mfinite8088** [sbtoonz/BackPacks_Remake#44]
Two of us play on a dedicated server.  If one of us is logged in and playing, backpack seems to work great.  We only experience issues when we're playing at the same time and both of us have backpacks equipped.

Tested last night at our base and if I tried to open my backpack while they have their backpack equipped, I would get a log entry on my client.

[Warning: Unity Log] 03/08/2022 19:57:41: Failed to find rpc method -553663219

If the other player switched back to wearing a cape, my backpack goes back to opening again flawlessly.

I can help test or whatever info you need that helps, just let me know.  Great mod.

**Comment by sbtoonz** [comment]
https://github.com/sbtoonz/BackPacks_Remake/releases/tag/0.3.11

Please test this version

**Comment by mfinite8088** [comment]
Update is installed on the server and both clients.  Still works great in solo.  If both of us are connected and in separate areas, it works fine.  If we're in the same area (like our base), neither of us can open the backpack.  This is what the log generates when we try.

[Info   : Unity Log] 03/09/2022 14:27:10: Setting selected recipe 0

[Info   : Unity Log] 03/09/2022 14:27:10: Setting selected recipe 0

It does the duplicate entry each time I try to open the backpack.

If either of us unequips the backpack and re-equip while next to each other, we can then open the backpack and it works fine until we separate and then come back to the same place.

Also, the rpc method log entry has not happened since testing this update.

**Comment by drew-id** [comment]
Installed on both server and clients. Also using Server Characters - not sure if this impacts how the backpacks work or not.

Can also confirm some abnormal behavior.

1) Removing some (not all) items from the backpack and when re-opening the bag all items were gone
NOTE: this is related to the above as this occurred in the presence of another player where both of us had issues opening the bags.

2) Vanishing Backpack items
- Disappearing when CTRL + CLICKING them into the backpack in specific but random slots. Replicated dragging items into the same (empty) slots where they also vanish. (Seems random, can't work out the trigger for this but with my friend when it occurs, so could be related)
- Closing/Opening bag or unequip/equipping does not fix the issue. However one time walking through the portal back to base, bag would not open. Unequip and equip and it works - only it now contains items (including Ore, a non-portal item and a duplicate of the Ore i removed from the bag previously - Duped Items).

3) Another vanishing items was when equipping a rugged backpack (non-upgraded) in replacement of my leather one (fully upgraded - more slots)
- The result is any items in the empty now unavailable slots vanished from my leather bag when re-quipping it after having the rugged one equipped.


All of the above occurred when around my friend, but when we experienced the issue our fix to get the backpacks working was different:

- ME - I had to unequip and re-equip the backpack
- FRIEND - he had to open ANY chest and then it his backpack would open

**Fixes noted in V0.2.4**
Resolved issue where if another player equipped or upgraded bag your bag would no longer open
Resolved issue where if YOU upgraded or unequipped your bag other players lost access to their bags


Happy to test any changes, replicate and video any abnormalities.
